include/circt/Dialect/RTGTest/IR/RTGTestAttributes.td

```diff
@@ -0,0 +1,60 @@
+//===- RTGTestAttributes.td - RTGTest attributes -----------*- tablegen -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// This describes the RTGTest attributes.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef CIRCT_DIALECT_RTGTEST_IR_RTGTESTATTRIBUTES_TD
+#define CIRCT_DIALECT_RTGTEST_IR_RTGTESTATTRIBUTES_TD
+
+include "mlir/IR/AttrTypeBase.td"
+include "mlir/IR/EnumAttr.td"
+
+// Flat allocation of unique IDs to all registers. The actual ID value does not
+// matter.
+def RegisterAttr : I32EnumAttr<
+  "Registers", "Unique IDs for all RTGTest registers", [
+    I32EnumAttrCase<"zero",  0>,
+    I32EnumAttrCase<"ra",  1>,
+    I32EnumAttrCase<"sp",  2>,
+    I32EnumAttrCase<"gp",  3>,
+    I32EnumAttrCase<"tp",  4>,
+    I32EnumAttrCase<"t0",  5>,
+    I32EnumAttrCase<"t1",  6>,
+    I32EnumAttrCase<"t2",  7>,
+    I32EnumAttrCase<"s0",  8>,
+    I32EnumAttrCase<"s1",  9>,
+    I32EnumAttrCase<"a0",  10>,
+    I32EnumAttrCase<"a1",  11>,
+    I32EnumAttrCase<"a2",  12>,
+    I32EnumAttrCase<"a3",  13>,
+    I32EnumAttrCase<"a4",  14>,
+    I32EnumAttrCase<"a5",  15>,
+    I32EnumAttrCase<"a6",  16>,
+    I32EnumAttrCase<"a7",  17>,
+    I32EnumAttrCase<"s2",  18>,
+    I32EnumAttrCase<"s3",  19>,
+    I32EnumAttrCase<"s4",  20>,
+    I32EnumAttrCase<"s5",  21>,
+    I32EnumAttrCase<"s6",  22>,
+    I32EnumAttrCase<"s7",  23>,
+    I32EnumAttrCase<"s8",  24>,
+    I32EnumAttrCase<"s9",  25>,
+    I32EnumAttrCase<"s10",  26>,
+    I32EnumAttrCase<"s11",  27>,
+    I32EnumAttrCase<"t3",  28>,
+    I32EnumAttrCase<"t4",  29>,
+    I32EnumAttrCase<"t5",  30>,
+    I32EnumAttrCase<"t6",  31>,
```

Comment:
This looks suspiciously like RISC-V :grinning:. Is the plan for the RTG dialect to at some point also "standardize" some of these things in CIRCT? Or are users of the RTG dialect expected to always provide their own ISA-describing ops?

Reply:
Well observed. The idea is that we add a description of RV32I or a subset of it to test the RTG infra. We chose an actual ISA because it allows us to have E2E integration tests upstream so that breaking changes to core RTG stuff are detected as early as possible. It *could* also serve as a small example of how to use the RTG infra, but it is not meant to "standardize" anything. Users are expected to come up with their own dialect describing what they want to generate random tests for.

Reply:
That makes sense!